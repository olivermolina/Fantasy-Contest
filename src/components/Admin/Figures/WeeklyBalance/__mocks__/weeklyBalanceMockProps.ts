export const weeklyBalanceMockData = [
  {
    agent: null,
    id: '1',
    username: 'jdoe',
    firstname: 'John',
    lastname: 'Doe',
    referral: '',
    isFirstDeposit: false,
    mondayBalance: 20,
    tuesdayBalance: 0,
    wednesdayBalance: -35,
    thursdayBalance: 0,
    fridayBalance: 0,
    saturdayBalance: 0,
    sundayBalance: 20,
    deposits: 50,
    withdrawals: 6,
    totalWeekBalance: 5,
    pendingTotal: 20,
    cashAmount: 0,
    withdrawable: 0,
    creditAmount: 0,
    totalBalance: 0,
    isActive: true,
  },
  {
    agent: null,
    id: '2',
    username: 'jwick',
    firstname: 'John',
    lastname: 'Wick',
    referral: '',
    isFirstDeposit: false,
    mondayBalance: 20,
    tuesdayBalance: 0,
    wednesdayBalance: -35,
    thursdayBalance: 0,
    fridayBalance: 0,
    saturdayBalance: 0,
    sundayBalance: 20,
    deposits: 50,
    withdrawals: 6,
    totalWeekBalance: 5,
    pendingTotal: 20,
    cashAmount: 0,
    withdrawable: 0,
    creditAmount: 0,
    totalBalance: 0,
    isActive: true,
  },
];
