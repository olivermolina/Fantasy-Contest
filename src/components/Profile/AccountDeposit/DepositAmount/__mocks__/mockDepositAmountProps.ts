export const mockDepositAmountProps = {
  setDepositAmount: () => {
    alert('Set Deposit amount');
  },
  handleNext: () => {
    alert('handleNext');
  },
  depositAmount: 10,
  isFirstDeposit: false,
  maxMatchDeposit: 50,
  depositAmountOptions: [10, 25, 50, 75, 100, 125],
  reloadBonusAmount: 0,
  reloadBonusType: 'FLAT',
  showBonusCreditStyles: false,
  minDepositAmount: 10,
};
