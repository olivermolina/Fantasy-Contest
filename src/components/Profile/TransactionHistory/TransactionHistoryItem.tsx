import React from 'react';
import { TransactionHistoryInterface } from '~/server/routers/user/transactionHistory';
import classNames from 'classnames';
import dayjs from 'dayjs';
import { EntryDatetimeFormat } from '~/constants/EntryDatetimeFormat';

interface Props {
  row: TransactionHistoryInterface;
}

export default function TransactionHistoryItem(props: Props) {
  const { row } = props;

  return (
    <div
      className={
        'grid grid-cols-2 lg:grid-cols-6 gap-2 lg:gap-4 bg-inherit p-4'
      }
    >
      <p className={'flex flex-col gap-2'}>
        <span className={'text-lightText text-sm lg:text-md'}>
          Transaction Date
        </span>
        <span className={'text-white text-md lg:text-lg'}>
          {dayjs(row.transactionDate).format(EntryDatetimeFormat)}
        </span>
      </p>
      <p className={'flex flex-col gap-2'}>
        <span className={'text-lightText text-sm lg:text-md'}>Amount</span>
        <span className={'text-white text-md lg:text-lg'}>
          {row.amount.toFixed(2)}
        </span>
      </p>
      <p className={'flex flex-col gap-2'}>
        <span className={'text-lightText text-sm lg:text-md'}>Type</span>
        <span className={'text-white text-md lg:text-lg'}>{row.type}</span>
      </p>
      <p className={'flex flex-col gap-2'}>
        <span className={'text-lightText text-sm lg:text-md'}>
          Free Credits
        </span>
        <span className={'text-white text-md lg:text-lg'}>
          {row.amountBonus.toFixed(2)}
        </span>
      </p>
      <p className={'flex flex-col gap-2'}>
        <span className={'text-lightText text-sm lg:text-md'}>Details</span>
        <span className={'text-white text-md lg:text-lg'}>{row.details}</span>
      </p>
      <p className={'flex flex-col gap-2'}>
        <span className={'text-lightText text-sm lg:text-md'}>Status</span>
        <div
          className={
            'flex w-fit text-md rounded-lg lg:text-lg bg-[#1A4265] items-center gap-2 p-2 '
          }
        >
          <span
            className={classNames('rounded-full p-2 w-2 h-2', {
              'bg-green-500': row.status === 'COMPLETE',
              'bg-yellow-600': row.status === 'PENDING',
              'bg-red-600': row.status === 'FAILED',
            })}
          />
          <p
            className={classNames('uppercase text-md text-white', {
              'text-green-500': row.status === 'COMPLETE',
              'text-yellow-600': row.status === 'PENDING',
              'text-red-600': row.status === 'FAILED',
            })}
          >
            {row.status}
          </p>
        </div>
      </p>
    </div>
  );
}
