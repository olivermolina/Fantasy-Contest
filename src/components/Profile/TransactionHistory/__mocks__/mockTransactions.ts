export const mockTransactions = [
  {
    id: 'ntTfe7gRlpn5jdFs',
    transactionDate: '2022-12-22T03:25:11.630Z',
    type: 'CREDIT',
    details: 'Cash Contest Cancelled',
    amount: 10,
    amountBonus: 0,
    status: 'COMPLETE',
    userId: 'ntTfe7gRlpn5jdFs',
    username: 'testuser',
  },
  {
    id: 'ethq8juazC7KtvgH',
    transactionDate: '2022-12-22T03:25:11.577Z',
    type: 'CREDIT',
    details: 'Win',
    amount: 25,
    amountBonus: 0,
    status: 'COMPLETE',
    userId: 'ntTfe7gRlpn5jdFs',
    username: 'testuser',
  },
  {
    id: 'UMI9tDWaT7xNtVHg',
    transactionDate: '2022-12-22T01:03:59.711Z',
    type: 'DEBIT',
    details: 'Place entry',
    amount: 10,
    amountBonus: 0,
    status: 'COMPLETE',
    userId: 'ntTfe7gRlpn5jdFs',
    username: 'testuser',
  },
  {
    id: 'wgbHYWZqSBtIctX8',
    transactionDate: '2022-12-21T05:27:23.935Z',
    type: 'DEBIT',
    details: 'Place entry',
    amount: 10,
    amountBonus: 0,
    status: 'COMPLETE',
    userId: 'ntTfe7gRlpn5jdFs',
    username: 'testuser',
  },
];
