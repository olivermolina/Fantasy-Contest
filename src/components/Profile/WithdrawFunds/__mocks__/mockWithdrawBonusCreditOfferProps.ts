export const mockWithdrawBonusCreditOfferProps = {
  handleSubmit: () => console.log('submit'),
  handleNext: () => console.log('next'),
  open: true,
  setOpen: () => console.log('set open'),
  maxRetentionBonus: 500,
  retentionBonusMatchMultiplier: 2,
  cashBalance: 100,
};
