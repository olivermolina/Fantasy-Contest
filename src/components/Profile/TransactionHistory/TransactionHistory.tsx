import React from 'react';
import type { TransactionHistoryInterface } from '~/server/routers/user/transactionHistory';
import TransactionHistoryItem from './TransactionHistoryItem';
import { LoadingSpinner } from '~/components/Cart/LoadingSpinner';

interface TransactionHistoryProps {
  isLoading: boolean;
  transactions: TransactionHistoryInterface[];
}

function TransactionHistory(props: TransactionHistoryProps) {
  return (
    <div
      className={
        'flex flex-col gap-2 divide-y divide-slate-500 border-y border-slate-500'
      }
    >
      {props.isLoading && (
        <div className="flex justify-center items-center p-4">
          <LoadingSpinner />
        </div>
      )}

      {!props.isLoading && props.transactions.length === 0 && (
        <div className="flex justify-center items-center p-4">
          No Transactions
        </div>
      )}

      {props.transactions.map((row) => (
        <TransactionHistoryItem key={row.id} row={row} />
      ))}
    </div>
  );
}

export default TransactionHistory;
