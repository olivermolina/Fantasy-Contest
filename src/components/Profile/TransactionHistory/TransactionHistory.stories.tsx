import React from 'react';
import { ComponentMeta, ComponentStory } from '@storybook/react';
import TransactionHistory from './TransactionHistory';
import { mockTransactions } from './__mocks__/mockTransactions';

// More on default export: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
export default {
  component: TransactionHistory,
  // More on argTypes: https://storybook.js.org/docs/react/api/argtypes
  argTypes: {
    backgroundColor: { control: 'color' },
  },
} as ComponentMeta<typeof TransactionHistory>;

// More on component templates: https://storybook.js.org/docs/react/writing-stories/introduction#using-args
const Template: ComponentStory<typeof TransactionHistory> = (args) => (
  <TransactionHistory {...args} />
);

export const Default = Template.bind({});
// More on args: https://storybook.js.org/docs/react/writing-stories/args
Default.args = {
  transactions: mockTransactions,
  isLoading: false,
};
