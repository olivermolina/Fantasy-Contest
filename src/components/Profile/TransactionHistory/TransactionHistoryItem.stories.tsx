import React from 'react';
import { ComponentMeta, ComponentStory } from '@storybook/react';
import TransactionHistoryItem from './TransactionHistoryItem';
import { mockTransactions } from './__mocks__/mockTransactions';

// More on default export: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
export default {
  component: TransactionHistoryItem,
  // More on argTypes: https://storybook.js.org/docs/react/api/argtypes
  argTypes: {
    backgroundColor: { control: 'color' },
  },
} as ComponentMeta<typeof TransactionHistoryItem>;

// More on component templates: https://storybook.js.org/docs/react/writing-stories/introduction#using-args
const Template: ComponentStory<typeof TransactionHistoryItem> = (args) => (
  <TransactionHistoryItem {...args} />
);

export const Item = Template.bind({});
// More on args: https://storybook.js.org/docs/react/writing-stories/args
Item.args = {
  row: mockTransactions[0],
};
