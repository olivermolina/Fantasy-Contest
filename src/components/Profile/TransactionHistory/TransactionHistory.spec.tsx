import React from 'react';
import { render } from '@testing-library/react';
import TransactionHistory from './TransactionHistory';
import { mockTransactions } from './__mocks__/mockTransactions';

describe('TransactionHistory', () => {
  it('should render correctly', () => {
    const props = {
      transactions: mockTransactions,
      isLoading: false,
    };

    const { container } = render(<TransactionHistory {...props} />);
    expect(container).toMatchSnapshot();
  });
});
