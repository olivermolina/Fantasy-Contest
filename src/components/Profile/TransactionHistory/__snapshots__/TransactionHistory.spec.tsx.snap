// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransactionHistory should render correctly 1`] = `
<div>
  <div
    class="flex flex-col gap-2 divide-y divide-slate-500 border-y border-slate-500"
  >
    <div
      class="grid grid-cols-2 lg:grid-cols-6 gap-2 lg:gap-4 bg-inherit p-4"
    >
      <p
        class="flex flex-col gap-2"
      >
        <span
          class="text-lightText text-sm lg:text-md"
        >
          Transaction Date
        </span>
        <span
          class="text-white text-md lg:text-lg"
        >
          Dec 22, 2022 03:25 AM
        </span>
      </p>
      <p
        class="flex flex-col gap-2"
      >
        <span
          class="text-lightText text-sm lg:text-md"
        >
          Amount
        </span>
        <span
          class="text-white text-md lg:text-lg"
        >
          10.00
        </span>
      </p>
      <p
        class="flex flex-col gap-2"
      >
        <span
          class="text-lightText text-sm lg:text-md"
        >
          Type
        </span>
        <span
          class="text-white text-md lg:text-lg"
        >
          CREDIT
        </span>
      </p>
      <p
        class="flex flex-col gap-2"
      >
        <span
          class="text-lightText text-sm lg:text-md"
        >
          Free Credits
        </span>
        <span
          class="text-white text-md lg:text-lg"
        >
          0.00
        </span>
      </p>
      <p
        class="flex flex-col gap-2"
      >
        <span
          class="text-lightText text-sm lg:text-md"
        >
          Details
        </span>
        <span
          class="text-white text-md lg:text-lg"
        >
          Cash Contest Cancelled
        </span>
      </p>
      <p
        class="flex flex-col gap-2"
      >
        <span
          class="text-lightText text-sm lg:text-md"
        >
          Status
        </span>
        <div
          class="flex w-fit text-md rounded-lg lg:text-lg bg-[#1A4265] items-center gap-2 p-2 "
        >
          <span
            class="rounded-full p-2 w-2 h-2 bg-green-500"
          />
          <p
            class="uppercase text-md text-white text-green-500"
          >
            COMPLETE
          </p>
        </div>
      </p>
    </div>
    <div
      class="grid grid-cols-2 lg:grid-cols-6 gap-2 lg:gap-4 bg-inherit p-4"
    >
      <p
        class="flex flex-col gap-2"
      >
        <span
          class="text-lightText text-sm lg:text-md"
        >
          Transaction Date
        </span>
        <span
          class="text-white text-md lg:text-lg"
        >
          Dec 22, 2022 03:25 AM
        </span>
      </p>
      <p
        class="flex flex-col gap-2"
      >
        <span
          class="text-lightText text-sm lg:text-md"
        >
          Amount
        </span>
        <span
          class="text-white text-md lg:text-lg"
        >
          25.00
        </span>
      </p>
      <p
        class="flex flex-col gap-2"
      >
        <span
          class="text-lightText text-sm lg:text-md"
        >
          Type
        </span>
        <span
          class="text-white text-md lg:text-lg"
        >
          CREDIT
        </span>
      </p>
      <p
        class="flex flex-col gap-2"
      >
        <span
          class="text-lightText text-sm lg:text-md"
        >
          Free Credits
        </span>
        <span
          class="text-white text-md lg:text-lg"
        >
          0.00
        </span>
      </p>
      <p
        class="flex flex-col gap-2"
      >
        <span
          class="text-lightText text-sm lg:text-md"
        >
          Details
        </span>
        <span
          class="text-white text-md lg:text-lg"
        >
          Win
        </span>
      </p>
      <p
        class="flex flex-col gap-2"
      >
        <span
          class="text-lightText text-sm lg:text-md"
        >
          Status
        </span>
        <div
          class="flex w-fit text-md rounded-lg lg:text-lg bg-[#1A4265] items-center gap-2 p-2 "
        >
          <span
            class="rounded-full p-2 w-2 h-2 bg-green-500"
          />
          <p
            class="uppercase text-md text-white text-green-500"
          >
            COMPLETE
          </p>
        </div>
      </p>
    </div>
    <div
      class="grid grid-cols-2 lg:grid-cols-6 gap-2 lg:gap-4 bg-inherit p-4"
    >
      <p
        class="flex flex-col gap-2"
      >
        <span
          class="text-lightText text-sm lg:text-md"
        >
          Transaction Date
        </span>
        <span
          class="text-white text-md lg:text-lg"
        >
          Dec 22, 2022 01:03 AM
        </span>
      </p>
      <p
        class="flex flex-col gap-2"
      >
        <span
          class="text-lightText text-sm lg:text-md"
        >
          Amount
        </span>
        <span
          class="text-white text-md lg:text-lg"
        >
          10.00
        </span>
      </p>
      <p
        class="flex flex-col gap-2"
      >
        <span
          class="text-lightText text-sm lg:text-md"
        >
          Type
        </span>
        <span
          class="text-white text-md lg:text-lg"
        >
          DEBIT
        </span>
      </p>
      <p
        class="flex flex-col gap-2"
      >
        <span
          class="text-lightText text-sm lg:text-md"
        >
          Free Credits
        </span>
        <span
          class="text-white text-md lg:text-lg"
        >
          0.00
        </span>
      </p>
      <p
        class="flex flex-col gap-2"
      >
        <span
          class="text-lightText text-sm lg:text-md"
        >
          Details
        </span>
        <span
          class="text-white text-md lg:text-lg"
        >
          Place entry
        </span>
      </p>
      <p
        class="flex flex-col gap-2"
      >
        <span
          class="text-lightText text-sm lg:text-md"
        >
          Status
        </span>
        <div
          class="flex w-fit text-md rounded-lg lg:text-lg bg-[#1A4265] items-center gap-2 p-2 "
        >
          <span
            class="rounded-full p-2 w-2 h-2 bg-green-500"
          />
          <p
            class="uppercase text-md text-white text-green-500"
          >
            COMPLETE
          </p>
        </div>
      </p>
    </div>
    <div
      class="grid grid-cols-2 lg:grid-cols-6 gap-2 lg:gap-4 bg-inherit p-4"
    >
      <p
        class="flex flex-col gap-2"
      >
        <span
          class="text-lightText text-sm lg:text-md"
        >
          Transaction Date
        </span>
        <span
          class="text-white text-md lg:text-lg"
        >
          Dec 21, 2022 05:27 AM
        </span>
      </p>
      <p
        class="flex flex-col gap-2"
      >
        <span
          class="text-lightText text-sm lg:text-md"
        >
          Amount
        </span>
        <span
          class="text-white text-md lg:text-lg"
        >
          10.00
        </span>
      </p>
      <p
        class="flex flex-col gap-2"
      >
        <span
          class="text-lightText text-sm lg:text-md"
        >
          Type
        </span>
        <span
          class="text-white text-md lg:text-lg"
        >
          DEBIT
        </span>
      </p>
      <p
        class="flex flex-col gap-2"
      >
        <span
          class="text-lightText text-sm lg:text-md"
        >
          Free Credits
        </span>
        <span
          class="text-white text-md lg:text-lg"
        >
          0.00
        </span>
      </p>
      <p
        class="flex flex-col gap-2"
      >
        <span
          class="text-lightText text-sm lg:text-md"
        >
          Details
        </span>
        <span
          class="text-white text-md lg:text-lg"
        >
          Place entry
        </span>
      </p>
      <p
        class="flex flex-col gap-2"
      >
        <span
          class="text-lightText text-sm lg:text-md"
        >
          Status
        </span>
        <div
          class="flex w-fit text-md rounded-lg lg:text-lg bg-[#1A4265] items-center gap-2 p-2 "
        >
          <span
            class="rounded-full p-2 w-2 h-2 bg-green-500"
          />
          <p
            class="uppercase text-md text-white text-green-500"
          >
            COMPLETE
          </p>
        </div>
      </p>
    </div>
  </div>
</div>
`;
